src/test/scala/io/iohk/ethereum/blockchain/sync/regular/BlockFetcherSpec.scala

```diff
@@ -124,6 +124,78 @@ class BlockFetcherSpec extends TestKit(ActorSystem("BlockFetcherSpec_System")) w
       peersClient.expectMsgClass(classOf[BlacklistPeer])
       peersClient.expectMsgPF() { case PeersClient.Request(msg, _, _) if msg == firstGetBlockHeadersRequest => () }
     }
+
+    "should ensure blocks passed to importer are always forming chain" in new TestSetup {
```

Comment:
Could this test be more similar to the ones above? I mean,  instead of invalidation, here we are mining block in the middle...I think this sequence should be enough:
- fetcher request headers
- we mine a new block (block 1  from a fork)
- fetcher should dismiss pending headers response
- fetcher should ask for new headers (from new top)
- fetcher should receive those blocks..